aspnet/fundamentals/middleware.rst

```diff
@@ -1,8 +1,176 @@
+
+You can see an example of setting up a request pipeline, using a variety of request delegates, in the default web site template that ships with Visual Studio 2015. Its ``Configure`` method, shown below, first wires up error pages (in development) or the site's production error handler, then builds out the pipeline with support for static files, ASP.NET Identity authentication, and finally, ASP.NET MVC.
+
+.. literalinclude:: ../../common/samples/WebApplication1/src/WebApplication1/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 89-134
+	:dedent: 8
+	:emphasize-lines: 11-13,19,23,26,36
+
+Because of the order in which this pipeline was constructed, the middleware configured by the ``UseErrorHandler`` method will catch any exceptions that occur in later calls (in non-development environments). Also, in this example a design decision has been made that static files will not be protected by any authentication. This is a tradeoff that improves performance when handling static files since no other middleware (such as authentication middleware) needs to be called when handling these requests (ASP.NET 5 uses a specific ``wwwroot`` folder for all files that should be accessible by default, so there is typically no need to perform authentication before sending these files). If the request is not for a static file, it will flow to the next piece of middleware defined in the pipeline (in this case, Identity). Learn more about :doc:`static-files`.
+
+.. note:: **Remember:** the order in which you arrange your ``Use[Middleware]`` statements in your application's ``Configure`` method is very important. Be sure you have a good understanding of how your application's request pipeline will behave in various scenarios.
+
+The simplest possible ASP.NET application sets up a single request delegate that handles all requests. In this case, there isn't really a request "pipeline", so much as a single anonymous function that is called in response to every HTTP request.
+
+.. literalinclude:: middleware/sample/src/MiddlewareSample/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 23-26
+	:dedent: 12
+
+It's important to realize that request delegate, as written here, will terminate the pipeline, regardless of other calls to ``App.Run`` that you may include. In the following example, only the first delegate ("Hello, World!") will be executed and displayed.
+
+.. literalinclude:: middleware/sample/src/MiddlewareSample/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 20-31
+	:emphasize-lines: 5
+	:dedent: 8
+
+You chain multiple request delegates together making a different call, with a ``next`` parameter representing the next delegate in the pipeline. Note that just because you're calling it "next" doesn't mean you can't perform actions both before and after the next delegate, as this example demonstrates:
```

Comment:
But beware of trying to modify the HttpResponse after calling next, as one of the following components most likely wrote to the response and caused it to be sent to the client.